lib/core/crypto.js

```diff
@@ -62,6 +62,10 @@ module.exports = (config) => {
     str,
     datatype
   ) => {
+    try {
+      http.reportMetric();
+    } catch (err) {}
```

Comment:
It's not good practice to swallow errors like this. Can we log something even if its at TRACE.

Reply:
This is done on purpose, we want the metric reporting to be completely silent for the user if it fails

Reply:
We definitely don't want to impair core functionality but It will make things difficult to debug in the future. Ideally we would log something at DEBUG or record an additional metric that we export once connectivity is restored. Even a comment explaining why we are swallowing the exception would be preferable.

Reply:
Added a comment, I don't think we should even add debug at trace - in the worst case we can add additional code when debugging ourselves